lib/karafka/connection/api_adapter.rb

```diff
@@ -84,17 +84,16 @@ def subscribe(topic)
         # @param topic [String] topic that we want to pause
         # @param partition [Integer] number partition that we want to pause
         # @param consumer_group [Karafka::Routing::ConsumerGroup] consumer group details
-        # @return [Array] array with all the details required to pause kafka consumer
+        # @return [Hash] hash with all the details required to pause kafka consumer
         def pause(topic, partition, consumer_group)
-          [
-            Karafka::App.config.topic_mapper.outgoing(topic),
-            partition,
-            {
+          {
+            args: [Karafka::App.config.topic_mapper.outgoing(topic), partition],
```

Comment:
Client could simply do something like `*args[0..-2], **args.last`, but I think that if there's an adapter, it should try and be more helpful with how to pass the params

Reply:
I agree :)